src/main/scala/cromwell/engine/ExecutionStore.scala

```diff
@@ -0,0 +1,37 @@
+package cromwell.engine
+
+import cromwell.binding.{Call, WdlBinding}
+import cromwell.engine.ExecutionStatus.ExecutionStatus
+import scala.collection.mutable
+
+object ExecutionStatus extends Enumeration {
+  type ExecutionStatus = Value
+  val NotStarted, Running, Failed, Done = Value
+}
+
+/**
+ * Corresponds to the "execution table" of our discussions.
+ */
+class ExecutionStore(binding: WdlBinding) {
+
+  def isDone: Boolean = !table.exists(_._2 != ExecutionStatus.Done)
+
+  def updateStatus(call: Call, status: ExecutionStatus.Value): Unit = {
+    table.put(call, status)
+  }
+
+  private def buildInitialTable =
+    mutable.Map[Call, ExecutionStatus]() ++
+      binding.workflow.calls.map { call => call -> ExecutionStatus.NotStarted }.toMap
+
+  val table: mutable.Map[Call, ExecutionStatus] = buildInitialTable
+
+  def runnableCalls: Iterable[Call] = {
+    val unstartedCalls = table.filter { case (_, status) => status == ExecutionStatus.NotStarted }.keys
+    unstartedCalls.filter { unstartedCall =>
+      val prereqCalls = unstartedCall.prerequisiteCalls().toSet
+      val donePrereqs = table.filter{ case (pCall, pStatus) => prereqCalls.contains(pCall) && pStatus == ExecutionStatus.Done}
+      donePrereqs.size == prereqCalls.size
```

Comment:
Yeah, this can be prettier.  If only there was some construct in Scala that rendered chains of `flatMap` and `filter` more readable... :smile: